core/modules/ccontrol/MergingRequester.cc

```diff
@@ -137,22 +145,16 @@ bool MergingRequester::flush(int bLen, bool last) {
         _state = MsgState::RESULT_WAIT;
         return true;
     case MsgState::RESULT_RECV:
-        LOGF_DEBUG("RESULT_RECV last = %1%" % last);
-        if (!last) {
-            _state = MsgState::BUFFER_DRAIN;
-            _buffer.resize(1);
-        }
-        return true;
-    case MsgState::BUFFER_DRAIN:
-        // The buffer should always be empty, but last is not always set to true by xrootd
-        // unless we ask xrootd to read at least one character.
-        LOGF_INFO("BUFFER_DRAIN last=%1% bLen=%2% buffer=%3%" %
-                last % bLen % util::prettyCharList(_buffer));
-        _buffer.resize(1);
-        return true;
+        // We shouldn't wind up here. _buffer.size(0) and last=true should end communication.
+        // fall-through
     case MsgState::HEADER_ERR:
     case MsgState::RESULT_ERR:
-        _setError(log::MSG_RESULT_ERROR, "Unexpected message");
+         {
+            std::ostringstream eos;
+            eos << "Unexpected message From:" << _wName << " flush state=" << getStateStr(_state) << " last=" << last;
+            LOGF_ERROR(eos.str().c_str());
```

Comment:
I'd replace it with `LOGF_ERROR("%1%" % eos.str());` just in case `_wName` can contain percent character in it.